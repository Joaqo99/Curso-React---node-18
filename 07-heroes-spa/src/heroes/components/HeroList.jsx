import { useMemo } from "react";
import { getHeroesBypublisher } from "../helpers/getHeroesByPublishiers";
import { HeroCard } from "./HeroCard";

export const HeroList = ({ publisher }) => {
    const heroes = useMemo(()=>getHeroesBypublisher( publisher), [publisher]) 
    return (
        <div className="row rows-cols-1 row-cols-md-3 g-3">
            {
                heroes.map(hero => 
                    <HeroCard key={hero.id} {...hero}/>
            )}
        </div>
    );
}
