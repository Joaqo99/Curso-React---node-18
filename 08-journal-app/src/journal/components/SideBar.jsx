import { TurnedInNot } from "@mui/icons-material";
import { Grid, Drawer, Typography, Toolbar, Divider, List, ListItem, ListItemButton, ListItemIcon, ListItemText } from "@mui/material";
import { Box } from "@mui/system";

export const SideBar = ({ drawerWidth }) => {
    return (
        <Box
            component="nav"
            sx={{width: {sm: drawerWidth}, flexShrink: {sm: 0}}}
        >

            <Drawer
                variant='permanent'
                open
                sx={{
                    display: { xs: 'block'},
                    '& .MuiDrawer-paper': {boxSizing: 'border-box', width: drawerWidth}
                }}
            >
                <Toolbar>
                    <Typography varient='h6' noWrap component='div'>Joaquin Varela</Typography>
                </Toolbar>
                <Divider />
                <List>
                    {
                        ['Enero', 'Febrero', 'Marzo', 'Abril'].map( text => (
                            <ListItem key={ text } disablePadding>
                                <ListItemButton>

                                    <ListItemIcon>
                                        <TurnedInNot />
                                    </ListItemIcon>

                                    <Grid container>
                                        <ListItemText primary={text} />
                                        <ListItemText secondary={'loremksmkldsamfkldsmafkdsfidskmksdlmcskdnms'} />
                                    </Grid>
                                </ListItemButton>
                            </ListItem>
                        ))
                    }
                </List>
            </Drawer>

        </Box>
    );
}