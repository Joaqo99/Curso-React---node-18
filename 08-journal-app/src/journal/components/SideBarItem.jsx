import { TurnedInNot } from "@mui/icons-material";
import { ListItem, ListItemButton, ListItemIcon, Grid, ListItemText, } from "@mui/material";
import { useMemo } from "react";
import { useDispatch } from "react-redux";
import { setActiveNote } from "../../store/journal/journalSlice";

export const SideBarItem = ({note}) => {

    const dispatch = useDispatch()

    const onClickNote = (note) => {
        dispatch(setActiveNote(note))
    }


    const newTitle = useMemo(()=>{
        return note.title.length > 15
            ? note.title.substring(0, 15) + '...'
            : note.title
    }, [note.title])


    return (
        <ListItem disablePadding>
            <ListItemButton onClick={() => onClickNote(note)}>

                <ListItemIcon>
                    <TurnedInNot />
                </ListItemIcon>

                <Grid container>
                    <ListItemText primary={newTitle} />
                    <ListItemText secondary={note.body} />
                </Grid>

            </ListItemButton>
        </ListItem>
    );
}