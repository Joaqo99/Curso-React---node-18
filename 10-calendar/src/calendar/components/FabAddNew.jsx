import { addHours } from "date-fns";
import { useCalendarStore } from "../../hooks/useCalendarStore";
import { useUiStore } from "../../hooks/useUiStore";

export const FabAddNew = () => {

    const {openDateModal} = useUiStore();
    const {setActiveEvent} = useCalendarStore();

    const handleClickNew = () => {
        setActiveEvent({
            title: '',
            notes: '',
            bgColor: '#fafafa',
            user: {
                _id: '123',
                name: 'Joaco'
            }
        })
        openDateModal();
    };

    return (
        <button
            className="btn btn-primary fab"
            onClick={handleClickNew}
        >
            <i className="fas fa-plus"></i>
        </button>
    );
}