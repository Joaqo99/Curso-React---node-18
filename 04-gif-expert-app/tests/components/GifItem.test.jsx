import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Pruebas en <GifItem />', ()=>{
    const title = 'Star Wars'
    const url = 'http://satrwars.com/darthvader.jpg'

    test('debe de hacer match con el snapshot', ()=>{
        const {container} = render(<GifItem title={title} url={url}/>)
        expect( container ).toMatchSnapshot()
    })

    test('debe de mostar la iamgen con el url y el alt indicado', ()=>{
        render(<GifItem title={title} url={url}/>)
        const {src, alt} = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(title)
    })

    test('debe de mostrar el titulo en el componente', ()=>{
        render(<GifItem title={title} url={url}/>)
        expect( screen.getByText( title )).toBeTruthy()
    })
})