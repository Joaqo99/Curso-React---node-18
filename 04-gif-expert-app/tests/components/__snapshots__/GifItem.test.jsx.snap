// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Star Wars"
      src="http://satrwars.com/darthvader.jpg"
    />
    <p>
      Star Wars
    </p>
  </div>
</div>
`;
